### ISSUE TITLE
Errors hitting console, but not reported

### ISSUE THREAD

**Issue by rainabba** [albanm/node-libxslt#58]
I'm trying to process a template that has XSL includes in it. Those files weren't available so the parser was spitting out errors on the console, but the returned err object only had, "Failed to parse stylesheet" in it which wasn't very helpful. It was a good 30 minutes before I saw the messages in the console and they were MUCH more helpful. This was especially nasty because I had no reason to think my template was bad. It is working elsewhere and well-formed.

Are those messages to stdout from the library and not available for you to include in err? What about passing a 3rd object into the callback and have it carry the full output from stdout?

**Comment by rainabba** [comment]
Turns out that I couldn't get imports/include working properly so I had to use https://github.com/misterion/gulp-import-xslt to pack my templates with their dependencies (which is fine for my environment);

Still haven't seen a proper transformation though (getting blank results). Here's hoping!

**Comment by albanm** [comment]
I am not a 100 percent sure, but I am quite confident that I did my best to provide error details.

I don't want to invest much time on this module. I try to answer questions and I gladly accept bug fixes in PRs so that it keeps working for its users, but not much more. So feel free to investigate :)

FYI when working with this module I regularly used xsltproc, a command line tool that is part of libxslt. It is a more direct way to test a transformation and it helps determine if a problem is in this module's code or not.

**Comment by rainabba** [comment]
Understood. I'll see what I can do to help. Found out my last issue has to do with the URI I use for the esxl namespace (for the sake of SQLXML 3.0, which includes the parser I'm moving away from). Once I fixed that URI to a more standard one, I got transforms working.

I may look into this and definitely won't hesitate to provide PRs. I have a similar project (took on a life of its own and I only help maintain it since that repo has become so popular. If you don't mind this sitting open for a bit, leave it and I'll reference, otherwise we can close and put it out of mind for now (hopefully to someone else's benefit).
